src/System.Net.Primitives/src/System/Net/IPv6AddressHelper.cs

```diff
@@ -311,7 +311,7 @@ internal static unsafe void Parse(string address, ushort* numbers, int start, re
                         for (++i; i < address.Length && address[i] != ']' && address[i] != '/'; ++i)
                         {
                         }
-                        scopeId = address.Substring(start, i - start);
+                        scopeId = new string(address.ToArray(), start, i - start);
```

Comment:
Ouch, we shouldn't need to allocate a char[] here.  string has a `string(ReadOnlySpan<char>)` ctor you should be able to use.